### PR TITLE
Fix FlashAttention3 optimal use when available

### LINKED ISSUE #196: [Bug] FlashAttention3 not employed when available & `attention_type` is `default`
In case of  `"attention_type": "default"`, `PytorchAttention` is used instead of   `FlashAttention3`. The issue  emerges from   `packages/ltx-core/src/ltx_core/model/transformer/attention.py`. 

A fix together with a fallback behavior is  provided in https://github.com/Lightricks/LTX-2/pull/195

### PR DESCRIPTION
**Pull request by 3manifold**
When the model cfg is set to `"attention_type": "default"` (e.g. see `config {"transformer": {"_class_name": "AVTransformer3DModel",...,"attention_type": "default", ...} }` in https://huggingface.co/Lightricks/LTX-2.3/blob/main/ltx-2.3-22b-dev.safetensors), `AttentionCallable` is not treated optimally in `packages/ltx-core/src/ltx_core/model/transformer/attention.py`.

In brief, `packages/ltx-core/src/ltx_core/model/transformer/model_configurator.py` passes down `attention_type` to `packages/ltx-core/src/ltx_core/model/transformer/attention.py`. There, even if the imports flag the various attention cases correctly (see `memory_efficient_attention`, `flash_attn_interface`) at the top of the file, `class AttentionFunction(Enum)` is mishandling `AttentionCallable` values.

This PR fixes that behaviour following the principle of `FA3 > XFormers > PyTorch`. It also adds a fallback from `FlashAttention3` to `PytorchAttention` in case of arbitrary attention masks to avoid errors during runs.

resolves https://github.com/Lightricks/LTX-2/issues/196

### CODE REVIEW

**Comment by michaellightricks** [comment]
we addressed the attention backend selection in our recent update, let us know if you have any feedback
